wagtail/admin/ui/side_panels.py

```diff
@@ -66,17 +67,22 @@ def get_context_data(self, parent_context):
         context = super().get_context_data(parent_context)
         user_perms = UserPagePermissionsProxy(self.request.user)
         page = self.object
+        lock = page.get_lock()
 
         if page.id:
             context.update(
                 {
                     "history_url": reverse(
                         "wagtailadmin_pages:history", args=(page.id,)
                     ),
+                    "lock": lock,
+                    "locked_for_user": lock is not None
+                    and lock.for_user(self.request.user),
```

Comment:
Not really sure about this, but probably can also be

```suggestion
                    "locked_for_user": page_perms.page_locked(),
```

with

```py
page_perms = user_perms.for_page(page)
```

somewhere before this?